modules/videoio/include/opencv2/videoio.hpp

```diff
@@ -719,6 +719,25 @@ enum VideoCaptureOBSensorProperties{
 
 //! @} videoio_flags_others
 
+/** @brief Read data stream interface
+ */
+class CV_EXPORTS_W IStreamReader
+{
+public:
+    virtual ~IStreamReader();
+
+    /** @brief Read bytes from stream */
+    virtual long long read(char* buffer, long long size) = 0;
+
+    /** @brief Sets the stream position
+     *
+     * @param offset Seek offset
+     * @param origin SEEK_SET / SEEK_END / SEEK_CUR
```

Comment:
Wouldn't it be better to completely avoid C/C++ constants in public interface and introduce our own?

Reply:
FFmpeg (basically) and Python use them too (with same values).
IMHO, it is not necessary to introduce extra conversions to/from own values.

Reply:
Agree, we can add such enum but potential user will still cast to integer or introduce long if/else/switch

Reply:
My concern was that user would be required to include `<stdio.h>`/`<cstdio>` in order to use this seemingly unrelated class. Though perhaps it's fine.